src/infinity/libraries/Bridges/LibCMARKCarbon.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+
+import "../LibRetire.sol";
+import "../Token/LibTransfer.sol";
+import "../../interfaces/ICMARKCredit.sol";
+import "../../C.sol";
+
+import "lib/forge-std/src/console.sol";
+
+/**
+ * @author MarcusAurelius
+ * @title LibCMARKCarbon
+ */
+
+library LibCMARKCarbon {
+    event CarbonRetired(
+        LibRetire.CarbonBridge carbonBridge,
+        address indexed retiringAddress,
+        string retiringEntityString,
+        address indexed beneficiaryAddress,
+        string beneficiaryString,
+        string retirementMessage,
+        address carbonToken,
+        uint retiredAmount
+    );
+
+    /**
+     * @notice                     Retire a CMARK project token

```

Comment:
`poolToken` param missing from natspec. if poolToken will never be needed though for `retireCmark` it might be clearer to remove it from the params altogether and just pass address(0) to the `saveRetirementDetails` poolToken arg.